### ISSUE TITLE
Can't Build on linux/Ubuntu 16.04

### ISSUE THREAD

**Issue by JackDevAU** [DeathCradle/Open-Terraria-API#23]
I have been successful with building and launching this on windows 10.
However I can't for the life of me build this without errors on an Ubuntu system.

I thought i was nearly there but then it couldn't find "Terraria.WindowsLaunch" in "GameRunner.cs"

This is my current process of building following [The Build Guide](https://github.com/DeathCradle/Open-Terraria-API/wiki/Building)
**Step 1.** Clone Master repo
**Step 2.** Place the **Windows** ReLogic.Native.dll and **Windows** TerrariaServer.exe in /wrap/TerrariaServer/ 
**Step 3.** Restore OTAPI.Server.sln Nuget Packages
**Step 4.** Build OTAPI.Server.sln 
**Step 5.**  Build OTAPI.Patcher
**Step 6.** Try Run OTAPI.Patcher.exe 
**Error**: 
Then i get this error when Trying to run The patcher.exe
I thought maybe trying to restore packages but this doesn't work either...
```
Unhandled Exception:
Mono.Cecil.AssemblyResolutionException: Failed to resolve assembly: 'Microsoft.Xna.Framework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=842cf8be1de50553'
  at Mono.Cecil.BaseAssemblyResolver.Resolve (Mono.Cecil.AssemblyNameReference name, Mono.Cecil.ReaderParameters parameters) [0x000f7] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.BaseAssemblyResolver.Resolve (Mono.Cecil.AssemblyNameReference name) [0x00007] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.DefaultAssemblyResolver.Resolve (Mono.Cecil.AssemblyNameReference name) [0x00025] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataResolver.Resolve (Mono.Cecil.TypeReference type) [0x00040] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.ModuleDefinition.Resolve (Mono.Cecil.TypeReference type) [0x00006] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.TypeReference.Resolve () [0x00006] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.Mixin.CheckedResolve (Mono.Cecil.TypeReference self) [0x00000] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.GetConstantType (Mono.Cecil.TypeReference constant_type, System.Object constant) [0x000a1] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.AddConstant (Mono.Cecil.IConstantProvider owner, Mono.Cecil.TypeReference type) [0x00007] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.AddField (Mono.Cecil.FieldDefinition field) [0x00072] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.AddFields (Mono.Cecil.TypeDefinition type) [0x00013] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.AddType (Mono.Cecil.TypeDefinition type) [0x0008c] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.AddTypeDefs () [0x00018] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.BuildTypes () [0x00014] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.BuildModule () [0x0007e] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.BuildMetadata () [0x00000] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.ModuleWriter.BuildMetadata (Mono.Cecil.ModuleDefinition module, Mono.Cecil.MetadataBuilder metadata) [0x00008] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.ModuleWriter.WriteModuleTo (Mono.Cecil.ModuleDefinition module, System.IO.Stream stream, Mono.Cecil.WriterParameters parameters) [0x000ab] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.ModuleDefinition.Write (System.IO.Stream stream, Mono.Cecil.WriterParameters parameters) [0x00022] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.ModuleDefinition.Write (System.String fileName, Mono.Cecil.WriterParameters parameters) [0x0000a] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.AssemblyDefinition.Write (System.String fileName, Mono.Cecil.WriterParameters parameters) [0x00000] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at ILRepacking.ILRepack.Repack () [0x00385] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at OTAPI.Patcher.Program.Main (System.String[] args) [0x0034e] in <169dec57a75c4694b3259e5d2bb859df>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: Mono.Cecil.AssemblyResolutionException: Failed to resolve assembly: 'Microsoft.Xna.Framework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=842cf8be1de50553'
  at Mono.Cecil.BaseAssemblyResolver.Resolve (Mono.Cecil.AssemblyNameReference name, Mono.Cecil.ReaderParameters parameters) [0x000f7] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.BaseAssemblyResolver.Resolve (Mono.Cecil.AssemblyNameReference name) [0x00007] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.DefaultAssemblyResolver.Resolve (Mono.Cecil.AssemblyNameReference name) [0x00025] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataResolver.Resolve (Mono.Cecil.TypeReference type) [0x00040] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.ModuleDefinition.Resolve (Mono.Cecil.TypeReference type) [0x00006] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.TypeReference.Resolve () [0x00006] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.Mixin.CheckedResolve (Mono.Cecil.TypeReference self) [0x00000] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.GetConstantType (Mono.Cecil.TypeReference constant_type, System.Object constant) [0x000a1] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.AddConstant (Mono.Cecil.IConstantProvider owner, Mono.Cecil.TypeReference type) [0x00007] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.AddField (Mono.Cecil.FieldDefinition field) [0x00072] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.AddFields (Mono.Cecil.TypeDefinition type) [0x00013] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.AddType (Mono.Cecil.TypeDefinition type) [0x0008c] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.AddTypeDefs () [0x00018] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.BuildTypes () [0x00014] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.BuildModule () [0x0007e] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.MetadataBuilder.BuildMetadata () [0x00000] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.ModuleWriter.BuildMetadata (Mono.Cecil.ModuleDefinition module, Mono.Cecil.MetadataBuilder metadata) [0x00008] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.ModuleWriter.WriteModuleTo (Mono.Cecil.ModuleDefinition module, System.IO.Stream stream, Mono.Cecil.WriterParameters parameters) [0x000ab] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.ModuleDefinition.Write (System.IO.Stream stream, Mono.Cecil.WriterParameters parameters) [0x00022] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.ModuleDefinition.Write (System.String fileName, Mono.Cecil.WriterParameters parameters) [0x0000a] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at Mono.Cecil.AssemblyDefinition.Write (System.String fileName, Mono.Cecil.WriterParameters parameters) [0x00000] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at ILRepacking.ILRepack.Repack () [0x00385] in <cb3fd49c3e8f4dc980bfa856efdddf05>:0 
  at OTAPI.Patcher.Program.Main (System.String[] args) [0x0034e] in <169dec57a75c4694b3259e5d2bb859df>:0 
```

**Comment by DeathCradle** [comment]
Microsoft.Xna.Framework, the assembly causing issues there, is a reference in terraria that probably doesn't exist on your ubuntu machine, whereas it probably exists in the GAC on your windows machine. I haven't tested patching on linux for a long time and not many do, but potentially for now manually copy the xna dlls across to the bin folder from windows to the linux machine - maybe with some luck that may work temporarily.

I'll keep this open to see what can be improved for patching on linux.

**Comment by JackDevAU** [comment]
Ok, cool. Used 7zip to extract the .msi to get the Microsoft.Xna.Framework.dll and that allowed me to build and run the Patcher! 

Now onto the next issue:
I am successful in building the OTAPI.Server.Tests.sln.
When trying to run the .exe this produced creates this error:

```
Unhandled Exception:
System.MissingFieldException: Field not found: bool Terraria.Main.SkipAssemblyLoad Due to: Could not find field in class
  at OTAPI.Tests.Program.Main (System.String[] args) [0x00001] in <14a9d4c5b839405e8ce3a1d596fd3dcd>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.MissingFieldException: Field not found: bool Terraria.Main.SkipAssemblyLoad Due to: Could not find field in class
  at OTAPI.Tests.Program.Main (System.String[] args) [0x00001] in <14a9d4c5b839405e8ce3a1d596fd3dcd>:0
```

Opening the project in VSCode and trying to build with dotnet produces this error:

```
Exception has occurred: CLR/System.BadImageFormatException
An unhandled exception of type 'System.BadImageFormatException' occurred in OTAPI.Server.Tests.NetCore.dll: 'Could not load file or assembly 'OTAPI, Version=1.4.0.3, Culture=neutral, PublicKeyToken=null'. An attempt was made to load a program with an incorrect format.'
   at OTAPI.Tests.Common.Server.Start(Boolean forceLoadAssembly, String[] args) in /home/cheese/Documents/1403/TerrariaAPIv2/OTAPI.Tests/OTAPI.Tests.Common/Server.cs:line 31
   at OTAPI.Server.Tests.NetCore.Program.Main(String[] args) in /home/cheese/Documents/1403/TerrariaAPIv2/OTAPI.Tests/OTAPI.Server.Tests.NetCore/Program.cs:line 9
```

**Comment by DeathCradle** [comment]
This should now fixed in 8b9086e
